oneandone/src/main/java/org/apache/jclouds/oneandone/rest/domain/Types.java

```diff
@@ -0,0 +1,220 @@
+            Map<Integer, ArchitectureType> map = new HashMap<Integer, ArchitectureType>();
+            ArchitectureType[] values = ArchitectureType.values();
+            for (ArchitectureType value : values) {
+                map.put(value.getId(), value);
+            }
+
+            return map;
+        }
+    }
+
+    public enum ImageFrequency {
+        ONCE("ONCE"),
+        DAILY("DAILY"),
+        WEEKLY("WEEKLY"),
+        UNRECOGNIZED("");
+
+        public static ImageFrequency fromValue(String v) {
+            return Enums.getIfPresent(ImageFrequency.class, v).or(UNRECOGNIZED);
+        }
+        // the value which is used for matching
+        // the json node value with this enum
+        private final String value;
+
+        ImageFrequency(final String type) {
+            value = type;
+        }
+
+        @Override
+        public String toString() {
+            return value;
+        }
+    }
+
+    public enum ServerState {
+        POWERING_ON("POWERING_ON"),
+        POWERING_OFF("POWERING_OFF"),
+        POWERED_ON("POWERED_ON"),
+        POWERED_OFF("POWERED_OFF"),
+        DEPLOYING("DEPLOYING"),
+        REBOOTING("REBOOTING"),
+        REMOVING("REMOVING"),
+        CONFIGURING("CONFIGURING"), UNRECOGNIZED("");
+
+        public static ServerState fromValue(String v) {
+            return Enums.getIfPresent(ServerState.class, v).or(UNRECOGNIZED);
+        }
+
+        // the value which is used for matching
+        // the json node value with this enum
+        private final String value;
+
+        ServerState(final String type) {
+            value = type;
+        }
+
+        @Override
+        public String toString() {
+            return value;
+        }
+    }
+
+    public enum ServerAction {
+        POWER_ON("POWER_ON"), POWER_OFF("POWER_OFF"), REBOOT("REBOOT");
+        // the value which is used for matching
+        // the json node value with this enum
+        private final String value;
+
+        ServerAction(final String type) {
+            value = type;
+        }
+
+        @Override
+        public String toString() {
+            return value;
+        }
+    }
+
+    public enum ServerActionMethod {
+        SOFTWARE("SOFTWARE"), HARDWARE("HARDWARE");
+        // the value which is used for matching
+        // the json node value with this enum
+        private final String value;
+
+        ServerActionMethod(final String type) {
+            value = type;
+        }
+
+        @Override
+        public String toString() {
+            return value;
+        }
+    }
+
+    public enum IPType {
+        IPV4("IPV4"),
+        IPV6("IPV6");
+        // the value which is used for matching
+        // the json node value with this enum
+        private final String value;
+
+        IPType(final String type) {
+            value = type;
+        }
+
+        @Override
+        public String toString() {
+            return value;
+        }
+    }
```

Comment:
Consider adding the `fromValue` to all enums and removing the unnecessary value when it is exactly the same as the enum name.